spring-kafka/src/main/java/org/springframework/kafka/support/DefaultKafkaHeaderMapper.java

```diff
@@ -324,12 +326,37 @@ else if (!(headerName.equals(JSON_TYPES)) && matchesForInbound(headerName)) {
 					populateJsonValueHeader(header, requestedType, headers);
 				}
 				else {
-					headers.put(headerName, headerValueToAddIn(header));
+					handleHeader(headerName, header, headers);
 				}
 			}
 		});
 	}
 
+	/**
+	 * Handle non-reserved headers in {@link DefaultKafkaHeaderMapper}.
+	 * @param headerName the header name.
+	 * @param header the header instance.
+	 * @param headers the target headers.
+	 * @since 4.0.0
+	 */
+	protected void handleHeader(String headerName, Header header, final Map<String, Object> headers) {
```

Comment:
This is mapping from Kafka headers, So, I believe this name is better in this context: `fromUserHeader`.
Since standard headers have been already mapped before.